io-sim-classes/src/Control/Monad/Class/MonadProbe.hs

```diff
@@ -60,3 +82,150 @@ instance MonadRunProbe IO IO where
+            , readLogTrace = fail "Not supported operation"
+            , traceStopping = fail "Not supported operation"
+            }
+
+logTraceToMonitoringFramework :: MonadIO m => m (LogTrace () m String)
+logTraceToMonitoringFramework = do
+    tr <- liftIO $ defaultConfigStdout >>= \conf -> setupTrace (Right conf) "ouroboros"
+    return $ LogTrace
+            { traceOutput = \a ->
+                -- t <- getMonotonicTime
+                liftIO $ logInfo tr $ pack a
+                -- logInfo tr $ pack $ "time: " ++ show t ++ ", payload: " ++ show a
+            , readLogTrace = fail "Not supported operation"
+            , traceStopping = liftIO $ shutdownTrace tr
+            }
+
+logTraceToMonitoringFrameworkExternalTrace
+    :: MonadIO m
+    => Trace m
+    -> m (LogTrace () m String)
+logTraceToMonitoringFrameworkExternalTrace tr = do
+    return $ LogTrace
+            { traceOutput = \a -> logInfo tr $ pack a
+            , readLogTrace = fail "Not supported operation"
+            , traceStopping = liftIO $ shutdownTrace tr
+            }
+
+withLoggingToLogObjectList
+    :: MonadTimer m
+    => (UTCTime, Time m)
+    -> (LogTrace [LogObject] m String -> m a)
+    -> m ([LogObject], a)
+withLoggingToLogObjectList startTime f = do
+  trace <- logTraceToLogObjectList startTime
+--   tr <- liftIO $ defaultConfigStdout >>= \conf -> setupTrace (Right conf) "ouroboros"
+--   _h <- fork $ forever $ readLogTrace trace >>= mapM_ (liftIO . (traceConditionally tr))
+  res <- f trace
+--   liftIO $ shutdownTrace tr
+--   kill h
+  msgs  <- readLogTrace trace
+  return (msgs, res)
+
+logTraceToLogObjectList :: MonadTimer m => (UTCTime, Time m) -> m (LogTrace [LogObject] m String)
+logTraceToLogObjectList (startUTC, startMonotonic) = do
+    c <- newTVarIO []
+    return $ LogTrace
+            { traceOutput = \a -> do
+                now <- toInt =<< getMonotonicTime
+                start <- toInt startMonotonic
+                let diff = realToFrac (fromIntegral (now - start) / (1000000::Double)) -- to seconds
+                -- when we will be able to use time >= 1.9.1
+                -- let diff = secondsToNominalDiffTime $ fromIntegral $ (now - start) * (1000000::Int) -- to picoseconds
+                t <- toThreadIdInt =<< myThreadId
+                let meta = LOMeta {
+                      tstamp = addUTCTime diff startUTC
+                    , tid    = t
+                    }
+                let content = LogMessage $ LogItem
+                                { liSelection = Both
+                                , liSeverity  = Info
+                                , liPayload   = pack a
+                                }
+                atomically $ modifyTVar' c ((LogObject meta content):)
+            , readLogTrace = atomically $ do
+                v <- readTVar c
+                writeTVar c []
+                return $ reverse v
+            , traceStopping = fail "Not supported operation"
+            }
```

Comment:
I'm not clear why all these utils live here in this package. Many of them are `MonadIO` and nothing to do with the classes in this package.

Reply:
> @coot Could you also suggest a better place to put `LogTrace` ?

> What about `Control.Monad.Logger` or something similar like `Control.Monad.Logger.Trace`. The former would clash with some libraries, so the latter is better.
https://github.com/input-output-hk/ouroboros-network/pull/298#issuecomment-464067480

Reply:
That's true; just use the `ouroboros-network` package for `Control.Monad.Logger`. This way any upstream dependency will have access to it. If you prefer it to be independent of `ouroboros-network` then you could add another package for that. That might be useful if at some point one will want to use `io-sim` and the logger without requiring `ouroboros-network`.

Reply:
I cannot be in `ouroboros-network` since tests that use the logging are in `typed-protocols`.

Reply:
You can also move the tests to `ouroboros-network`.

Reply:
I have added some logging messages  in `runPeer` which also lives in the lib of `typed-protocols`.